salt/modules/slsutil.py

```diff
@@ -33,3 +36,87 @@
 
     salt '*' slsutil.merge '{foo: Foo}' '{bar: Bar}'
 '''
+
+
+def renderer(path=None, string=None, default_renderer='jinja|yaml', **kwargs):
+    '''
+    Parse a string or file through Salt's renderer system
+
+    This is an open-ended function and can be used for a variety of tasks. It
+    makes use of Salt's "renderer pipes" system to run a string or file through
+    a pipe of any of the loaded renderer modules.
+
+    :param path: The path to a file on the filesystem.
+    :param string: An inline string to be used as the file to send through the
+        renderer system. Note, not all renderer modules can work with strings;
+        the 'py' renderer requires a file, for example.
+    :param default_renderer: The renderer pipe to send the file through; this
+        is overridden by a "she-bang" at the top of the file.
+    :param kwargs: Keyword args to pass to Salt's compile_template() function.
+
+    Keep in mind the goal of each renderer when choosing a render-pipe; for
+    example, the Jinja renderer processes a text file and produces a string,
+    however the YAML renderer processes a text file and produces a data
+    structure.
+
+    One possible use is to allow writing "map files", as are commonly seen in
+    Salt formulas, but without tying the renderer of the map file to the
+    renderer used in the other sls files. In other words, a map file could use
+    the Python renderer and still be included and used by an sls file that uses
+    the default 'jinja|yaml' renderer.
+
+    For example, the two following map files produce identical results but one
+    is written using the normal 'jinja|yaml' and the other is using 'py':
+
+    .. code-block:: jinja
```

Comment:
you have ``salt['mod.fun']`` and ``salt.mod.fun`` here, wouldn't it make sense to use only one of them?